src/cameras/frame_grabber.py

```diff
@@ -2,26 +2,39 @@
 import threading
 import time
 import traceback
-
-import cv2
+from pathlib import Path
+from uuid import uuid4
 
 from src.cameras.dto import FramePayload
+from src.cameras.dtos.create_writer_options import get_canonical_time_str
+from src.cameras.video_writer.video_writer import SaveOptions, VideoWriter
+from src.config.data_paths import freemocap_data_path
 
 logger = logging.getLogger(__name__)
 
 
 class FrameThread(threading.Thread):
-    def __init__(self, get_next_frame, writer: cv2.VideoWriter, save_video=False):
+    def __init__(
+        self,
+        webcam_id: str,
+        get_next_frame,
+        frame_width: int,
+        frame_height: int,
+        save_video=False,
+    ):
         super().__init__()
+        self._webcam_id = webcam_id
+        self._frame_width = frame_width
+        self._frame_height = frame_height
         self._save_video = save_video
         self._is_capturing_frames = False
         self._get_next_frame = get_next_frame
         self._num_frames_processed = 0
         self._elapsed = 0
         self._frame: FramePayload = FramePayload()
+        self._session_id = uuid4()
+        self._session_timestr = get_canonical_time_str()
```

Comment:
It also gets a timestamp using the time_str function discussed above. That way, we can denote a "session"